### ISSUE TITLE
enum class expected to have a marshaller function in generated kotlin classes

### ISSUE THREAD

**Issue by akhudek** [apollographql/apollo-android#1901]
**Summary**
I'm encountering generated kotlin code that tries to call ".marshaller()" on a generated enum class. The enum class doesn't have this function though and thus I guess a compiler error.

**Description**
Here is the code that won't compile:

```kotlin
// AUTO-GENERATED FILE. DO NOT MODIFY.
//
// This class was automatically generated by Apollo GraphQL plugin from the GraphQL queries it found.
// It should not be modified by hand.
//
package com.myapp.plantbot.type

import com.apollographql.apollo.api.Input
import com.apollographql.apollo.api.InputFieldMarshaller
import com.apollographql.apollo.api.InputType
import kotlin.Suppress
import kotlin.collections.List

/**
 * on conflict condition type for table "plants"
 */
@Suppress("NAME_SHADOWING", "UNUSED_ANONYMOUS_PARAMETER", "LocalVariableName",
    "RemoveExplicitTypeArguments", "NestedLambdaShadowedImplicitParameter")
data class Plants_on_conflict(
  val constraint: Plants_constraint,
  val update_columns: List<Plants_update_column>,
  val where: Input<Plants_bool_exp> = Input.absent()
) : InputType {
  override fun marshaller(): InputFieldMarshaller = InputFieldMarshaller { writer ->
    writer.writeObject("constraint", constraint.marshaller())
    writer.writeList("update_columns") { listItemWriter ->
      update_columns.forEach { value ->
        listItemWriter.writeObject(value.marshaller())
      }
    }
    if (where.defined) writer.writeObject("where", where.value?.marshaller())
  }
}
```
And for example `Plants_constraint` is defined like:

```kotlin
// AUTO-GENERATED FILE. DO NOT MODIFY.
//
// This class was automatically generated by Apollo GraphQL plugin from the GraphQL queries it found.
// It should not be modified by hand.
//
package  com.myapp.plantbot.type

import kotlin.String

/**
 * unique or primary key constraints on table "plants"
 */
enum class Plants_constraint(
  val rawValue: String
) {
  /**
   * unique or primary key constraint
   */
  PLANTS_PKEY("plants_pkey"),

  /**
   * Auto generated constant for unknown enum values
   */
  UNKNOWN__("UNKNOWN__");

  companion object {
    fun safeValueOf(rawValue: String): Plants_constraint = values().find { it.rawValue == rawValue }
        ?: UNKNOWN__
  }
}
```
Note that it has no `marshaller()` method.

**Version**
I'm using `'com.apollographql.apollo:apollo-runtime:1.2.2'` and `'com.apollographql.apollo:apollo-gradle-plugin:1.2.2'`.

**Comment by sav007** [comment]
That's weird as we have test fixture for exactly the same case:

https://github.com/apollographql/apollo-android/blob/master/apollo-compiler/src/test/graphql/com/example/input_object_type/type/ReviewInput.kt#L48

https://github.com/apollographql/apollo-android/blob/master/apollo-compiler/src/test/graphql/com/example/input_object_type/type/ReviewInput.kt#L115

Could you please double check that `Plants_on_conflict` input type has field `constraint` that is enum in your `schema.json` (file with GraphQL schema introspection query).

**Comment by akhudek** [comment]
I think this is the subset of the schema you're looking for right? 
```json
       {
        "kind": "INPUT_OBJECT",
        "name": "plants_on_conflict",
        "description": "on conflict condition type for table \"plants\"",
        "fields": null,
        "inputFields": [
          {
            "name": "constraint",
            "description": null,
            "type": {
              "kind": "NON_NULL",
              "name": null,
              "ofType": {
                "kind": "ENUM",
                "name": "plants_constraint",
                "ofType": null,
                "__typename": "__Type"
              },
              "__typename": "__Type"
            },
            "defaultValue": null,
            "__typename": "__InputValue"
          },
          {
            "name": "update_columns",
            "description": null,
            "type": {
              "kind": "NON_NULL",
              "name": null,
              "ofType": {
                "kind": "LIST",
                "name": null,
                "ofType": {
                  "kind": "NON_NULL",
                  "name": null,
                  "ofType": {
                    "kind": "ENUM",
                    "name": "plants_update_column",
                    "ofType": null,
                    "__typename": "__Type"
                  },
                  "__typename": "__Type"
                },
                "__typename": "__Type"
              },
              "__typename": "__Type"
            },
            "defaultValue": null,
            "__typename": "__InputValue"
          },
          {
            "name": "where",
            "description": null,
            "type": {
              "kind": "INPUT_OBJECT",
              "name": "plants_bool_exp",
              "ofType": null,
              "__typename": "__Type"
            },
            "defaultValue": null,
            "__typename": "__InputValue"
          }
        ],
        "interfaces": null,
        "enumValues": null,
        "possibleTypes": null,
        "__typename": "__Type"
      },
```

**Comment by Daviziu** [comment]
Just encountered the same issue. Seems like @akhudek and me both are trying to access a Hasura GraphQL server. I get the same error on another Input Type named `…_on_conflict` where the generated type tries to call `marshaller()` on an Enum.

I tried replicating that error with a minimal custom GraphQL Schema, but everything I tried always worked correctly.

However, I do have a simple Hasura-based schema.json and an operation which will result in the error in question. I attached a .zip file with the two files in question.

[apollo-android-hasura-marshaller-bug.zip](https://github.com/apollographql/apollo-android/files/4239422/apollo-android-hasura-marshaller-bug.zip)
